### PR TITLE
ddl: support basic row count stat for distributed adding index via ETCD

### LINKED ISSUE #43642: support basic row count stat for distributed add index
We can use ETCD to save the row count stat. It would be removed and improved after the distributed framework support stat collection.

### PR DESCRIPTION
**Pull request by wjhuang2016**
### What problem does this PR solve?

Issue Number: close #43642

Problem Summary:

### What is changed and how it works?

This PR supports basic row count stat for distributed adding index via ETCD.
Every scheduler save the row count stat to etcd every 5 seconds.
DDL would sum up these stats from ETCD.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
image

- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: ddl/scheduler.go [hunk 1]
```diff
@@ -100,9 +106,53 @@ func NewBackfillSchedulerHandle(taskMeta []byte, d *ddl) (scheduler.Scheduler, e
...
+		if err != nil {
+			return nil, err
+		}
+		bh.totalRowCnt = int64(cnt)
+	}
+
+	bh.done = make(chan struct{})
+	go func() {
+		bh.UpdateStatLoop()
+	}()
```
Comment (tangenta) [inline ddl/scheduler.go:129]:
```suggestion
	go bh.UpdateStatLoop()
```

File: ddl/reorg.go [hunk 2]
```diff
@@ -134,6 +136,44 @@ func (rc *reorgCtx) getRowCount() int64 {
...
+
+func gcETCDRowCntStatIfNecessary(ctx context.Context, reorgInfo *reorgInfo, job *model.Job, client *clientv3.Client) {
+	if reorgInfo.Job.ReorgMeta.IsDistReorg && !reorgInfo.mergingTmpIdx {
+		path := fmt.Sprintf("distAddIndex/%d", job.ID)
+		_, err := client.Delete(ctx, path, clientv3.WithPrefix())
+		if err != nil {
+			logutil.BgLogger().Warn("[ddl] delete row count from ETCD failed", zap.Error(err))
+		}
+	}
+}
```
Comment (tangenta) [inline ddl/reorg.go:175]: The function name is nothing related to the body. It should be `cleanDistributedRowCount()`.

File: ddl/reorg.go [hunk 3]
```diff
@@ -134,6 +136,44 @@ func (rc *reorgCtx) getRowCount() int64 {
 	return row
 }
 
+func getDistAddIndexRowCntAndSetJobRowCnt(ctx context.Context, reorgInfo *reorgInfo, rc *reorgCtx, job *model.Job, client *clientv3.Client) int64 {
```
Comment (tangenta) [inline ddl/reorg.go:139]: Renaming it into `setJobRowCount` is better, because not only distributed row count is updated.

**Review by tangenta — APPROVED** [review]

File: ddl/reorg.go [hunk 4]
```diff
@@ -134,6 +136,44 @@ func (rc *reorgCtx) getRowCount() int64 {
 	return row
 }
 
+func getAndSetJobRowCnt(ctx context.Context, reorgInfo *reorgInfo, rc *reorgCtx, job *model.Job, client *clientv3.Client) int64 {
+	rowCount := int64(0)
+	if reorgInfo.Job.ReorgMeta.IsDistReorg && !reorgInfo.mergingTmpIdx {
+		path := fmt.Sprintf("distAddIndex/%d", job.ID)
```
Comment (zimulala) [inline ddl/reorg.go:142]: Create a generic variable for "distAddIndex"?

File: ddl/reorg.go [hunk 5]
```diff
@@ -134,6 +136,44 @@ func (rc *reorgCtx) getRowCount() int64 {
...
+		rowCount = rc.getRowCount()
+		job.SetRowCount(rowCount)
+	}
+	return rowCount
+}
+
+func deleteETCDRowCntStatIfNecessary(ctx context.Context, reorgInfo *reorgInfo, job *model.Job, client *clientv3.Client) {
+	if reorgInfo.Job.ReorgMeta.IsDistReorg && !reorgInfo.mergingTmpIdx {
+		path := fmt.Sprintf("distAddIndex/%d", job.ID)
+		_, err := client.Delete(ctx, path, clientv3.WithPrefix())
```
Comment (zimulala) [inline ddl/reorg.go:170]: Do we need do some retry?

File: ddl/scheduler.go [hunk 6]
```diff
@@ -103,17 +110,61 @@ func NewBackfillSchedulerHandle(taskMeta []byte, d *ddl) (scheduler.Scheduler, e
...
+	response, err := d.etcdCli.Get(ctx, path)
+	if err != nil {
+		return err
+	}
+	if len(response.Kvs) > 0 {
+		cnt, err := strconv.Atoi(string(response.Kvs[0].Value))
+		if err != nil {
+			return err
+		}
+		b.totalRowCnt = int64(cnt)
```
Comment (zimulala) [inline ddl/scheduler.go:162]: Is this the only place `totalRowCnt` is updated? Don't we need to update it when this subtask finished?

**Review by zimulala — APPROVED** [review]
LGTM

**Pull request merged**